armi/reactor/reactors.py

```diff
@@ -111,6 +111,40 @@ def add(self, container):
                 )
             self.core = cores[0]
 
+    def incrementAssemNum(self):
+        """
+        Increase the max assembly number by one and return the current value.
+
+        Returns
+        -------
+        int
+            The new max Assembly number.
+        """
+        val = int(self.p.maxAssemNum)
+        self.p.maxAssemNum += 1
+        return val
+
+    def normalizeNames(self):
+        """
+        Renumber and rename all the Assemblies and Blocks.
+
+        This method normalizes the names in the Core then the SFP.
+
+        Returns
+        -------
+        int
+            The new max Assembly number.
+        """
+        self.p.maxAssemNum = 0
+
+        ind = self.core.normalizeNames(self.p.maxAssemNum)
+        self.p.maxAssemNum = ind
```

Comment:
why is this line done twice in sequence? Seems kinda confusing.

Reply:
The first is `self.core` and the second is `self.sfp`.

As stated in the docstring.  The method does one first, and then the second, in an attempt to get all the Children (and other assemblies) in the Reactor.